scaladoc/src/test/scala/pink/cozydev/protosearch/scaladoc/ScaladocSuite.scala

```diff
@@ -0,0 +1,219 @@
+  /** This is a Scaladoc comment
+      * This is a description for the object Main.
+      */
+    object Main {
+          /** This is a class representing a person
+      * @param name The name of the person
+      * @param age The age of the person
+      */
+
+      /** 
+        * This function sums two integers.
+        * @param a The first parameter
+        * @param b The second parameter
+        * @tparam T The type parameter
+        */
+      @deprecated("Use add instead", "1.0")
+      def sum[T](a: Int, b: Int=1): Int = a + b
+
+         /** 
+        * This function greets the user.
+        * @param name The name parameter
+        */
+      @throws(classOf[IllegalArgumentException])  
+      def greet(name: String="Charles"): Unit = println(s"Hello, $name!")
+
+      /** 
+        * This function subtracts two integers.
+        * @param c The first parameter to subtract
+        * @param d The second parameter to subtract
+        * @tparam T The type parameters
+        */
+        implicit val myvalue:Int=3
+      def subtraction[T]( c: Int=21)(implicit d: Int): Int = c - d
+       /** 
+        * This function subtracts two integers.
+        * @tparam A The type parameters
+        * @tparam B The type parameters
+        */
+    }
+    """
+  val parseTestSum = """
+  object Main {
+    /**
+      * This function sums two integers.
+      * @param a The first parameter
+      * @param b The second parameter
+      * @tparam T The type parameter
+      */
+    @deprecated("Use add instead", "1.0")
+    def sum[T](a: Int, b: Int=1): Int = a + b
+  }
+  """
+  val parseTestfromList = """
+  package pink.cozydev.protosearch
+
+import scala.collection.mutable.ListBuffer
+
+/** An intermediate helper for iterating over documents and building an Index */
+final case class IndexBuilder[A] private (
+    fieldAndGetters: List[(Field, A => String)],
+    defaultField: String,
+) {
+  val schema = Schema(fieldAndGetters.head._1, fieldAndGetters.tail.map(_._1))
+
+  def fromList(docs: List[A]): MultiIndex = {
+    val fields = fieldAndGetters.map(_._1)
+    val buffers: Map[String, ListBuffer[List[String]]] =
+      fields.map(k => (k.name, ListBuffer.empty[List[String]])).toMap
+    val storage: Map[String, ListBuffer[String]] =
+      fields.map(k => (k.name, ListBuffer.empty[String])).toMap
+
+    docs.foreach { doc =>
+      fieldAndGetters.foreach { case (field, getter) =>
+        val fieldValue = getter(doc)
+        storage(field.name) += fieldValue
+        buffers(field.name) += field.analyzer.tokenize(fieldValue)
+      }
+    }
+    val indexes = fields.map { f =>
+      val idx =
+        if (f.positions) PositionalIndex(buffers(f.name).toList)
+        else FrequencyIndex(buffers(f.name).toList)
+      (f.name, idx)
+    }.toMap
+    new MultiIndex(
+      indexes = indexes,
+      schema = schema,
+      fields = storage.map { case (k, v) => (k, v.toArray) },
+    )
+  }
+}
+  
+  """
```

Comment:
This test input has a lot of code but very little scaladoc that we're actually testing. Could we perhaps shrink it down?